src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/UpdateAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,212 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.Sql.Common;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+using System.Security;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    /// <summary>
+    /// Defines the Update-AzureRmSqlManagedInstance cmdlet
+    /// </summary>
+    [Cmdlet(VerbsData.Update, "AzureRmSqlManagedInstance",
+        SupportsShouldProcess = true,
+        ConfirmImpact = ConfirmImpact.Medium)]
+    public class UpdateAzureSqlManagedInstance : ManagedInstanceCmdletBase
+    {
+        protected const string UpdateByNameAndResourceGroupParameterSet =
+            "UpdateManagedInstanceFromInputParameters";
+
+        protected const string UpdateByInputObjectParameterSet =
+            "UpdateManagedInstanceFromAzureSqlManagedInstanceModelInstanceDefinition";
+
+        protected const string UpdateByResourceIdParameterSet =
+            "UpdateManagedInstanceFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the Managed instance to use.
+        /// </summary>
+        [Parameter(ParameterSetName = UpdateByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "SQL Database Managed instance name.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedInstanceName { get; set; }
```

Comment:
Rename this to Name, and switch order of the parameters to ResourceGroupName then Name.